haystack/nodes/reader/table.py

```diff
@@ -32,6 +32,70 @@
 logger = logging.getLogger(__name__)
 
 
+class BaseTableProcessor:
+    def preprocess(self, docs: List[Document]):
+        return docs
+
+    def postprocess(self, answers: List[Answer], docs: List[Document]):
+        return answers
+
+
+class TableProcessor(BaseTableProcessor):
+    def __init__(self, num_rows=7, num_cols=7, num_row_fixed=0, num_col_fixed=0) -> None:
+        self.num_rows = num_rows
+        self.num_cols = num_cols
+        self.num_row_fixed = num_row_fixed
+        self.num_col_fixed = num_col_fixed
+
+    def preprocess(self, docs: List[Document]):
+        split_docs = []
+        for doc in docs:
+            table = doc.content
+            n_row, n_col = table.shape
+            for r in range(0, n_row, self.num_rows):
+                for c in range(0, n_col, self.num_cols):
+                    table_subset = table.iloc[r : r + self.num_rows, c : c + self.num_cols].reset_index(drop=True)
+                    split_docs.append(
+                        Document(
+                            content=table_subset,
+                            content_type="table",
+                            id=doc.id,
+                            meta={"row_offset": r, "col_offset": c},
+                        )
+                    )
+        return split_docs
+
+    def _remap_offset(self, answer: Answer, doc: Document):
+        # Return offset in split table document to the original parent document space
+        row_offset = doc.meta.get("row_offset", 0)
+        col_offset = doc.meta.get("col_offset", 0)
+
+        new_offsets = []
+
+        for offset in answer.offsets_in_document:
+            # TODO: Support Span in addition to TableCell; also, look at offsets_in_context
+            new_offsets.append(TableCell(row=row_offset + offset.row, col=col_offset + offset.col))
+
+        answer.offsets_in_document = new_offsets
+        return answer
+
+    def postprocess(self, answers: List[Answer], docs: List[Document]):
+        """Keep one answer per original document and adjust the offset of the answer."""
```

Comment:
I think we should combine both answers from the `split_documents` into a single answer. Some things to consider:
- If the `aggregator` is the same between both answers. 
- If the `aggregator` is different then I would either choose only one answer that has the higher score, or return two answers for the document.

Reply:
Oh yes, missed the fact that some aggregation would ideally run across splits!

To support it properly, might have to do this postprocessing prior to the aggregation being run though I think would be the case mainly for AVERAGE where we can't do average of averages.

Reply:
Great point about the AVERAGE! This probably means updating the `postprocess` method of the `_TableQuestionAnsweringPipeline` class to support this.